internal/kubeclient/scheme.go

```diff
@@ -0,0 +1,61 @@
+package kubeclient
+
+import (
+	argocd "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/pkg/errors"
+	authzv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubescheme "k8s.io/client-go/kubernetes/scheme"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+)
+
+var kargoSchemes = runtime.SchemeBuilder{
+	corev1.AddToScheme,
+	kargoapi.AddToScheme,
+}
+
+func newScheme(extraSchemes ...func(*runtime.Scheme) error) (*runtime.Scheme, error) {
+	scheme := runtime.NewScheme()
+	for _, addToScheme := range extraSchemes {
+		if err := addToScheme(scheme); err != nil {
+			return nil, errors.Wrap(err, "add api to scheme")
+		}
+	}
+	return scheme, nil
+}
+func NewAPIScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
```

Comment:
This function is used by two packages:

* `internal/api`
* `internal/kubeclient/manifest`

In `internal/kubeclient/manifest` it is used only to test the parser, which is only used in `internal/api` -- meaning the parser probably should live in `internal/api` as well.

If the parser moves to `internal/api`, it's viable to use a package-private helper function to DRY up references to this particular scheme.